### PR TITLE
Add FAQ about Cody supporting license for private and GPL code

### PR DESCRIPTION
**Pull request by MaedahBatool**
The PR adds an FAQ regarding Cody's support for private and GPL-licensed code. 

## Reference

- https://sourcegraph.slack.com/archives/C01DXLN3D0T/p1709760264845289

### CODE REVIEW

File: docs/cody/faq.mdx [hunk 1]
```diff
@@ -25,6 +25,10 @@ Yes, Cody is compatible with self-hosted Sourcegraph instances. However, there a
 - For certain repositories, Cody may utilize embeddings, which involves sending repository data to another third-party service like OpenAI
 - To use Cody effectively, your self-hosted Sourcegraph instance must have internet access for these interactions with external services
 
+### Is Cody licensed for private code, and does it allow GPL-licensed code?
+
+There are no checks or exclusions for Cody PLG (VS Code, JetBrains, Neovim) for private and GPL-licensed code. We are subject to whatever the LLMs are trained on. However, using [StarCoder for autocomplete](/cody/clients/enable-cody-enterprise#use-starcoder-for-autocomplete) in PLG is trained on permissively licensed code.
```
Comment (aramaraju) [inline docs/cody/faq.mdx:30]: "However, using [StarCoder for autocomplete](/cody/clients/enable-cody-enterprise#use-starcoder-for-autocomplete) in PLG is trained on permissively licensed code." 

The sentence feels off to me. What do you think about:

"However, Cody can be used with StarCoder for autocomplete which is trained only on permissively licensed code. "

We can then link to some external doc on how StarCoder is trained on permissive code. Thoughts?

**Review by aramaraju — COMMENTED** [review]
Let's clean up the text before merging the changes

File: docs/cody/faq.mdx (hunk 1 above)
Comment (MaedahBatool) [reply docs/cody/faq.mdx:30]: @aramaraju external as far as some official link is available I guess we can link to it.

Comment (aramaraju) [reply docs/cody/faq.mdx]: Thoughts on this one:
https://huggingface.co/bigcode/starcoder#%23model-summary 
In their Attribution section, they mention permissive code.

**Pull request merged**